src/fields/field_common.cc

```diff
@@ -38,7 +38,8 @@ FieldCommon::FieldCommon()
 
 
 FieldCommon::FieldCommon(const FieldCommon & other)
-: shared_(other.shared_),
+: name_(other.name_),
```

Comment:
I think that omitting copy of the name_ is intentional (unfortunately not documented). 
E.g. it is used in Field<..>::copy_from. Why you need it with the name_ copy?
To add name_ copy, we must first somehow find all places where this constructor is used.
Definitely, we have to fix its documentation and explain why we do not copy name_.